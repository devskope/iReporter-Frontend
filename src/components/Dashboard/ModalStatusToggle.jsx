import React from 'react';
import PropTypes from 'prop-types';

const ModalStatusToggles = ({ statusChangeHandler }) => (
  <div className="detail-modal__toggle-block">
    <h4 className="detail-modal__sub">Status</h4>
    <select
      className="detail-modal__status-toggle"
      onChange={statusChangeHandler}
    >
      <option value="draft">Draft</option>
      <option value="under investigation">Under Investigation</option>
      <option value="rejected">Rejected</option>
      <option value="resolved">Resolved</option>
    </select>
  </div>
);

ModalStatusToggles.propTypes = {
  statusChangeHandler: PropTypes.func.isRequired,
};

export default ModalStatusToggles;
