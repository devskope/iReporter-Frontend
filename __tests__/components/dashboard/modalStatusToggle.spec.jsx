import React from 'react';
import { mount } from 'enzyme';
import ModalStatusToggle from '../../../src/components/Dashboard/ModalStatusToggle';

const setup = () => {
  const props = {
    statusChangeHandler: jest.fn(),
  };
  return mount(<ModalStatusToggle {...props} />);
};

describe('ModalStatusToggle', () => {
  it('should render without crashing', () => {
    const wrapper = setup();

    expect(wrapper.find('.detail-modal__toggle-block').exists()).toBe(true);
    expect(wrapper.find('.detail-modal__sub').exists()).toBe(true);
    expect(wrapper.find('.detail-modal__status-toggle').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
