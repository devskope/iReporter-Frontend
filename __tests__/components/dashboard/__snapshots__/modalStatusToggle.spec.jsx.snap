// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalStatusToggle should match snapshot 1`] = `ReactWrapper {}`;
